File: model/src/main/java/io/github/runtimemodels/chazm/entity/EntityModule.java
package io.github.runtimemodels.chazm.entity;

import com.google.inject.AbstractModule;
import com.google.inject.assistedinject.FactoryModuleBuilder;
import io.github.runtimemodels.chazm.id.IdModule;

/**
 * The {@linkplain EntityModule} class provides a Guice binding module for entities.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public class EntityModule extends AbstractModule {

    @Override
    protected void configure() {
        install(new IdModule());
        final FactoryModuleBuilder factoryModuleBuilder = new FactoryModuleBuilder().implement(Agent.class, AgentEntity.class)
                .implement(Attribute.class, AttributeEntity.class).implement(Capability.class, CapabilityEntity.class)
                .implement(Characteristic.class, CharacteristicEntity.class).implement(InstanceGoal.class, InstanceGoalEntity.class)
                .implement(Pmf.class, PmfEntity.class).implement(Policy.class, PolicyEntity.class).implement(Role.class, RoleEntity.class)
                .implement(SpecificationGoal.class, SpecificationGoalEntity.class);
        install(factoryModuleBuilder.build(AgentFactory.class));
        install(factoryModuleBuilder.build(AttributeFactory.class));
        install(factoryModuleBuilder.build(CapabilityFactory.class));
        install(factoryModuleBuilder.build(CharacteristicFactory.class));
        install(factoryModuleBuilder.build(InstanceGoalFactory.class));
        install(factoryModuleBuilder.build(PmfFactory.class));
        install(factoryModuleBuilder.build(PolicyFactory.class));
        install(factoryModuleBuilder.build(RoleFactory.class));
        install(factoryModuleBuilder.build(SpecificationGoalFactory.class));
        install(factoryModuleBuilder.build(EntityFactory.class));
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/EntityModule.kt
package runtimemodels.chazm.model.entity

import com.google.inject.AbstractModule
import com.google.inject.assistedinject.FactoryModuleBuilder
import runtimemodels.chazm.api.entity.*
import runtimemodels.chazm.model.id.IdModule

/**
 * The [EntityModule] class provides a Guice binding module for entities.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
class EntityModule : AbstractModule() {

    override fun configure() {
        install(IdModule())
        val factoryModuleBuilder = FactoryModuleBuilder()
            .implement(Agent::class.java, AgentEntity::class.java)
            .implement(Attribute::class.java, AttributeEntity::class.java)
            .implement(Capability::class.java, CapabilityEntity::class.java)
            .implement(Characteristic::class.java, CharacteristicEntity::class.java)
            .implement(InstanceGoal::class.java, InstanceGoalEntity::class.java)
            .implement(Pmf::class.java, PmfEntity::class.java)
            .implement(Policy::class.java, PolicyEntity::class.java).implement(Role::class.java, RoleEntity::class.java)
            .implement(SpecificationGoal::class.java, SpecificationGoalEntity::class.java)
        install(factoryModuleBuilder.build(AgentFactory::class.java))
        install(factoryModuleBuilder.build(AttributeFactory::class.java))
        install(factoryModuleBuilder.build(CapabilityFactory::class.java))
        install(factoryModuleBuilder.build(CharacteristicFactory::class.java))
        install(factoryModuleBuilder.build(InstanceGoalFactory::class.java))
        install(factoryModuleBuilder.build(PmfFactory::class.java))
        install(factoryModuleBuilder.build(PolicyFactory::class.java))
        install(factoryModuleBuilder.build(RoleFactory::class.java))
        install(factoryModuleBuilder.build(SpecificationGoalFactory::class.java))
        install(factoryModuleBuilder.build(EntityFactory::class.java))
    }

}
